### ISSUE TITLE
Tern validator is triggered for projects that are not Tern projects

### ISSUE THREAD

**Issue by mniewrzal** [angelozerr/tern.java#418]
Projects that aren't converted to Tern projects are also processed by validator. This I believe is not necessary.

**Comment by angelozerr** [comment]
@mnieiwrzal indeed a WTP document region processor is added when the editor is opened. See https://github.com/angelozerr/tern.java/blob/master/eclipse/jsdt/tern.eclipse.ide.jsdt.ui/src/tern/eclipse/ide/jsdt/internal/ui/JavaEditorTracker.java#L165

I have commented the test with "project has tern nature" (it means that it contains a .tern-project file in the project, I don't use Eclipse nature) because when your editor is opened and you convert your project (or create at hand the .tern-project), a document region processor must be added to benefit with validation.

To do that with clean mean we should:

 * add document region processor only if project has tern nature.
 * add document region processor if the project is converted to tern project.

Any contribution are welcome!

**Comment by angelozerr** [comment]
@mniewrzal could you tell me the problem that you have please (when you do Build, it execute Tern Validator even for project which dosn't contains .tern-project?)

**Comment by mniewrzal** [comment]
Sorry for delayed response. Yes, when project is being built Tern validator is visible in Progress view. Project doesn't contain .tern-project. I'm not 100% sure that validation is fully performed but at least its visible. See screenshot.

![screenshot from 2016-07-08 09-04-18](https://cloud.githubusercontent.com/assets/7313031/16679949/d9c7a5dc-44eb-11e6-82d2-13804da14b75.png)

Eclipse Neon PHP EPP + Tern feature 1.2.0.201606160856

**Comment by angelozerr** [comment]
@mniewrzal you should have the same problem with HTML Validator no?

If you see https://github.com/angelozerr/tern.java/blob/master/eclipse/linters/tern.eclipse.ide.linter.core/src/tern/eclipse/ide/linter/internal/core/validation/TernValidator.java#L85, the validator checks that it's a tern project (a project which contains a .tern-project file in the root of the project).

**Comment by mniewrzal** [comment]
Ok, then I think this is how WTP validation framework shows progress. For me the issue can be closed.

**Comment by angelozerr** [comment]
@mniewrzal you are right, today the whole js resources of the project try to be validated even if the project is not a tern project:

 * in case of the project is a tern project, JS files are validated.
 * in case of the project is NOT a tern project, JS files are NOT validated.

But it should be better not to loop for each resources. To do that, WTP provides a filter rule, but it works only with project nature, facet. In tern.java I don't use project nature. WST Validation must be improved to provide a mean to implement custom filter. I have created a bug at https://bugs.eclipse.org/bugs/show_bug.cgi?id=497681

**Comment by mniewrzal** [comment]
Thanks for reporting this issue to WTP :)
